lc_macro_pipeline/macro_pipeline.py

```diff
@@ -78,12 +79,33 @@ def _run_task(f):
     #     bag = dask.bag.from_delayed(delayed_tasks)
     #     return bag.compute()
 
+    def setup_client(self, 
+                     mode = 'auto',
+                     address='0.0.0.0', 
+                     num_workers=1, 
+                     port=0, 
+                     num_threads_per_worker=1, 
+                     memory_limit=0,
+                     silence_logs=False):
+        if mode == 'auto':
+            self.client = Client()
+        elif mode == 'mannual':
+            localcluster = LocalCluster(
+                ip=address,
+                scheduler_port=port,
+                n_workers=num_workers,
+                memory_limit=memory_limit,
+                threads_per_worker=num_threads_per_worker,
+                silence_logs=silence_logs
+            )
+            self.client = Client(localcluster)
```

Comment:
The default client settings `client = Client()` seem to be equivalent to `cluster = LocalCluster() ; client = Client(cluster)`. So maybe we can exploit it to simplify this (other modes could be other types of clusters, like `SLURMCluster` etc.):

```suggestion
        if mode == 'local':
            cluster = LocalCluster(*args, **kwargs)
        elif mode == ...:
            clutster = ...
        self.client = Client(localcluster)
```